chronos/timer.nim

```diff
@@ -8,45 +8,397 @@
+  ## Returns ``true`` if ``a`` less or equal ``b``.
+  result = (a.value <= b.value)
+
+func `>`*(a, b: Duration): bool {.inline.} =
+  ## Returns ``true`` if ``a`` bigger then ``b``.
+  result = (a.value > b.value)
+
+func `>`*(a, b: Moment): bool {.inline.} =
+  ## Returns ``true`` if ``a`` bigger then ``b``.
+  result = (a.value > b.value)
+
+func `>=`*(a, b: Duration): bool {.inline.} =
+  ## Returns ``true`` if ``a`` bigger or equal ``b``.
+  result = (a.value >= b.value)
+
+func `>=`*(a, b: Moment): bool {.inline.} =
+  ## Returns ``true`` if ``a`` bigger or equal ``b``.
+  result = (a.value >= b.value)
+
+func `*`*(a, b: Duration): Duration {.inline.} =
+  result.value = a.value * b.value
+
+func `div`*(a, b: Duration): Duration {.inline.} =
+  result.value = a.value div b.value
+
+const
+  Nanosecond* = Duration(value: 1'i64)
+  Microsecond* = Duration(value: 1_000'i64) * Nanosecond
+  Millisecond* = Duration(value: 1_000'i64) * Microsecond
+  Second* = Duration(value: 1_000'i64) * Millisecond
+  Minute* = Duration(value: 60'i64) * Second
+  Hour* = Duration(value: 60'i64) * Minute
+  Day* = Duration(value: 24'i64) * Hour
+  Week* = Duration(value: 7'i64) * Day
+
+  ZeroDuration* = Duration(value: 0'i64)
+  InfiniteDuration* = Duration(value: -1'i64)
+
+func nanoseconds*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with nanoseconds value ``v``.
+  result.value = cast[int64](v)
+
+func microseconds*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with microseconds value ``v``.
+  result.value = cast[int64](v) * Microsecond.value
+
+func milliseconds*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with milliseconds value ``v``.
+  result.value = cast[int64](v) * Millisecond.value
+
+func seconds*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with seconds value ``v``.
+  result.value = cast[int64](v) * Second.value
+
+func minutes*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with minutes value ``v``.
+  result.value = cast[int64](v) * Minute.value
+
+func hours*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with hours value ``v``.
+  result.value = cast[int64](v) * Hour.value
+
+func days*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with days value ``v``.
+  result.value = cast[int64](v) * Day.value
+
+func weeks*(v: SomeInteger): Duration {.inline.} =
+  ## Initialize Duration with weeks value ``v``.
+  result.value = cast[int64](v) * Week.value
+
+func nanoseconds*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to nanoseconds.
+  result = v
+
+func microseconds*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to microseconds.
+  result = (v div Microsecond) * Microsecond
+
+func milliseconds*(v: Duration): Duration {.inline.} =
+  ## Round Duration ``v`` to milliseconds.
+  result = (v div Millisecond) * Millisecond
+
+func seconds*(v: Duration): Duration {.inline.} =
```

Comment:
think it might be useful to return `int64` here and elsewhere, so that it's possible to interface with other systems.. 
two overloads, so we can use them in each direction:
 `func seconds(v: Duration): int64 = ..` and `func seconds(v: int64): Duration` 

consider for example: `echo seconds(10).milliseconds()` - I want this to print 10000, and give me an integer I can serialize for example

Reply:
`echo toInt(seconds(10), Millisecond)`